storage/reader.go

```diff
@@ -443,3 +444,9 @@ func (mrd *MultiRangeDownloader) Wait() {
 func (mrd *MultiRangeDownloader) GetHandle() []byte {
 	return mrd.reader.getHandle()
 }
+
+// ValidateOpen validates that the MultiRangeDownloader is open and can be used.
+// It returns an error if the stream is permanently closed.
+func (mrd *MultiRangeDownloader) ValidateOpen() error {
```

Comment:
I still am not crazy about this name... Maybe we should just call this Error()? This will become more accurate once we make this get a meaningful error to clarify why there was a termination.